Sources/TensorFlow/Operators/Basic.swift

```diff
@@ -88,7 +101,11 @@ public extension Tensor {
     @inlinable
     @differentiable(vjp: _vjpSplit(count:alongAxis:) where Scalar: TensorFlowFloatingPoint)
     func split(count: Int, alongAxis axis: Int = 0) -> [Tensor] {
-        _Raw.split(splitDim: Tensor<Int32>(Int32(axis)), value: self, numSplit: Int64(count))
+        preconditionAxis(axis)
+        precondition(
+            shapeTensor[axis].scalarized() % Int32(count) == 0,
```

Comment:
Note, I think this call to `.scalarized()` will truncate lazy-tensor style traces, as well as prevent dynamic graph extraction. I believe @bgogul was working on some "Swift-side" shape tracking. Do you know the status of this @bgogul ? Could that help us here?

Reply:
The swift-side shape tracking is already in: https://github.com/tensorflow/swift-apis/blob/275fe8837f9a6e8e5af53dea8052be87a8a5a39d/Sources/TensorFlow/Core/LazyTensorOperation.swift#L78

This should not adversely affect the lazy tensor trace extraction.